src/utils/formUtils.js

```diff
@@ -0,0 +1,27 @@
+
+export const labels = {
+  bio: 'Bio',
+  email: 'E-mail',
+  phone: 'Phone',
+
+};
+
+export const placeHolders = {
+  fullName: 'Full name',
+  description: 'Description',
+  phone: '+XXX XXX XXX XXX',
+  email: 'E-mail',
+};
+
+export function validateInput(name, value, validationFunc, errors) {
```

Comment:
V čem spočívá výhoda předávání objektu `errors` v parametru fce? Musí se vytvářet pak i proměnná `newErrors`? Co kdybych pak metodu zavolal tímto způsobem? (falseValidator je jen testovací validátor, který vrací vždy false)

```.js
const falseValidator = () => false

let errors = {}
validateInput('fieldName', 3, falseValidator, errors)
errors['x'] = false;

```
Co bude obsahovat proměnná `errors` a proč?

Reply:
jako pipe, flow, kde se do metody posial aktualni stav chyb, tedy je jen pto jedno formularove pole ale pro vsechny a zadna validacni funkce neni zodpovedna za stav chyb ktere se ji netykaji

Reply:
cele to umoznuje vetis reuse, validate, by slo parametrizovat, nap telefon nemusi byt povinny atd., zlo by validate volat od jinud a jen pro nektere fieldy atd.

Reply:
A jaké hodnoty by nabývala ta proměnná `errors` z příkladu? Díky za odpověď

Reply:
z jakeho prikladu? 

zkusim to vysvetlit:
1) validateInput parazituje na objektu errors, nevyrabi novy pokud existuje, a jen pridava validani flag (podle name)
2) mozna te mate zapis:
  let newErrors = errors;
to je jen reassignemtn aby konvenoval s eslint nastavenim (AirBnB)
 
export function validateInput(name, value, validationFunc, errors) {
  const valid = validationFunc(value);
  let newErrors = errors;
  if (!valid) {
    newErrors = newErrors || {};
    newErrors[name] = true;
  } else if (valid && newErrors) {
    newErrors[name] = null;
    delete newErrors[name];
  }
  return newErrors;
}

Reply:
Myslel jsem tento příklad 
```.js
const falseValidator = () => false

let errors = {}
validateInput('fieldName', 3, falseValidator, errors)
errors['x'] = false;
```

jaká bude hodnota proměnné `errors` a proč?

Reply:
``` {x: false, fieldName: false}
fieldName
:
false
x
:
false
__proto__
:
Object ```

Reply:
prvni jej naplnis validateInput('fieldName', 3, falseValidator, errors)  (za predpokladu ze nezapomenes ';' nebo pouzijes standardjs :-))
podruhe errors['x'] = false;

Reply:
Poslední dotaz :) Kdybych příklad upravil a výstup fce uložil do proměnné `newErrors`.

```.js
const falseValidator = () => false

let errors = {}
const newErrors = validateInput('fieldName', 3, falseValidator, errors)
errors['x'] = false;
```

co bude obsahovat proměnná `newErrors` až kód doběhne? A proč?

Reply:
moc nevim kam otakza miri, ale protoze je to pruchozi reference,
newError bude to same jako v predchozim pripade
newErrors === errors, mutujeme ve validateInput
```
{fieldName: true, x: false}
```